chalice/deploy/planner.py

```diff
@@ -686,6 +686,41 @@ def _plan_s3bucketnotification(self, resource):
             ),
         ]
 
+    def _plan_cloudwatchevent(self, resource):
+        # type: (models.CloudWatchEvent) -> Sequence[InstructionMsg]
+
+        function_arn = Variable(
+            '%s_lambda_arn' % resource.lambda_function.resource_name
+        )
+
+        plan = [
+            models.APICall(
+                method_name='get_or_create_rule_arn',
+                params={'rule_name': resource.rule_name,
+                        'event_pattern': resource.event_pattern},
```

Comment:
It would be great if we could consolidate a good portion of this logic into a single shared method with the `_plan_scheduledevent` method. Looking over the code, really the only difference between these two functions are the use of `event_pattern` as a parameter instead of `schedule_event`.